index.js

```diff
@@ -1,8 +1,142 @@
+    })
+    archive.pipe(output)
+    results = uploadDirectory + '/ligand_out.pdbqt'
+    archive.append(fs.createReadStream(results), { name: 'ligand_out.pdbqt' })
+    log = uploadDirectory + '/log.txt'
+    archive.append(fs.createReadStream(log), { name: 'log.txt' })
+    archive.finalize();
+
+});
+
+app.post('/v1/autodock', (req, res) => {
+    ligand = null;
+    macromolecule = null;
+    fields = {};
+    errorMsg = "";
+    jobId = crypto.createHmac('SHA256', crypto.randomBytes(48))
+    .update(Date.now()
+    .toString())
+    .digest('hex');
+    if (!fs.existsSync(__dirname + '/uploads/')) {
+        fs.mkdirSync(__dirname + '/uploads/');
+    }
+    uploadDirectory = __dirname + '/uploads/' + jobId;
+    if (!fs.existsSync(uploadDirectory)) {
+        fs.mkdirSync(uploadDirectory);
+    }
+    else {
+        res.status(400);
+        errorMsg = 'Hash collision.';
+    }
+    var form = new formidable.IncomingForm();
+    form.multiples = true;
+    form.parse(req);
+    form.on('field', function(name, value) {
+        fields[name] = value
+    })
+    form.on('fileBegin', function (name, file){
+        if (fs.existsSync(uploadDirectory + name + '.pdbqt')) {
+            res.status(400);
+            errorMsg = 'Multiple files with same name uploaded?';
+        }
+        if (name == 'ligand') {
+            ligand = file.name        
+            file.path = uploadDirectory + '/' + 'ligand.pdbqt';
+        }
+        else if (name == 'macromolecule') {
+            macromolecule = file.name
+            file.path = uploadDirectory + '/' + 'macromolecule.pdbqt';
+        }
+        else {
+            res.status(400);
+            errorMsg = 'Unknown file name parameter: ' + name;
+        }
+
+    });
+    form.on('end', function() {
+        if (errorMsg) {
+            res.send(errorMsg);
+        }
+        try {
+            args = 
+            ['--receptor ', uploadDirectory + '/macromolecule.pdbqt',  
+            '--ligand ', uploadDirectory + '/ligand.pdbqt',
+            '--center_x', fields['center_x'],
+            '--center_y', fields['center_y'],
+            '--center_z', fields['center_z'],
+            '--size_x', fields['size_x'],
+            '--size_y', fields['size_y'],
+            '--size_z', fields['size_z'],
+            '--log', uploadDirectory + '/log.txt']
+        }
+        catch(err) {
+            res.status(400)
+            errorMsg = 'Incorrect arguments provided.';
+        }
+        try {
+            exec(`${__dirname}/vina `, args, {shell: true}, function(error, stdout, stderr) {
+                if (stderr) {
+                    console.log(stderr);
+                    // Make this create error.txt
+                }
+            });
+            response = {
+                'jobId': jobId,
+                'macromolecule': macromolecule,
+                'ligand': ligand
+            }
+            res.status(200);
+            res.send(response);
+        }
+        catch(error) {
```

Comment:
Per the documentation [here](https://nodejs.org/api/child_process.html) and [here](https://nodejs.org/api/child_process.html#child_process_class_childprocess) about child process, error is only returned if the actual process isn't started or can't be killed. It doesn't carry any error conditions or non-zero exit codes (so any actual issues with Vina).

We should definitely add something to the `if (stderr)` block, to throw an error there. 
Currently a search space of 0 gives a usage error that just silently kills the job, without ever telling the user.